modules/users/client/components/ProfileViewBasics.component.js

```diff
@@ -1,21 +1,213 @@
 import React from 'react';
-import { NamespacesConsumer } from 'react-i18next';
+import classnames from 'classnames';
+import { withNamespaces } from '@/modules/core/client/utils/i18n-angular-load';
+import { Trans } from 'react-i18next';
 import '@/config/lib/i18n';
 import PropTypes from 'prop-types';
+import * as languages from '@/config/languages/languages';
+import { hasConnectedAdditionalSocialAccounts, isWarmshowersId, socialAccountLink } from './utils/networks';
 
-export default function ProfileViewBasics({ profile }) {
-  return (<NamespacesConsumer ns="user-profile">{ t => (<>
+
+export function ProfileViewBasics({ t, profile }) {
+
+  /*
+  * Functions passing strings to translation fuction for translation scripts
+  */
+  const getGender = (genderCode) => (
+    t(genderCode)
```

Comment:
Translation string extractor won't pick this up, because it's a variable and it doesn't know what's in that variable. In Angular version, we weren't too concerned beacause we didn't have i18n set up.

Previous version (https://github.com/Trustroots/trustroots/commit/7a5053050e96cf5fc2f63c3fb49b1b62a60ece55#diff-5be83c916d70e68382e672abc55aa996L15) of this function wasn't that great either because it had partial strings meant to be placed in between sentences, while in some languages the whole sentence should be translated differently depending on gender. More over it didn't seem like we were using sentence version anywhere, so it was too early optimization.

I'd suggest having this:

```js
  const getGender = (genderCode) => (
    switch (genderCode) {
      case 'female':
        return t('Female');
      case 'male':
        return t('Male');
      case 'non-binary':
        return t('Non-binary');
      case 'other':
        return t('Other');
      default:
        return null; // React expects `null` for non-rendering
    }
  };
```

Does that make sense?

Reply:
As a side note, is there a way to define "context" for strings here so that it's easier to translate?

something like `t_with_context('Other', 'gender');`

Reply:
Looks like I suggested the previous structure myself. :-) https://github.com/Trustroots/trustroots/pull/1000#discussion_r243780448

And now I see that we had "sentence" when there is birthdate: _"33 years, male."_ — since it was such a simple sentence, I didn't even think it in the traditional sense of "sentence". 🤔

Reply:
I was not sure how 'text-capitalize' will work with languages other than English. Today I checked it briefly and it seemed fine.

Reply:
@simison 
> As a side note, is there a way to define "context" for strings here so that it's easier to translate?
> something like `t_with_context('Other', 'gender');`

Do you have an example of what you mean?